File: 118-pascals-triangle/pascals-triangle.ts
function generate(numRows: number): number[][] {
    let ans : number [][] = [];
    ans.push([1]);

    for(let i=0;i<numRows - 1;i++)
    {
        let prev:number = 0;
        let arr:number[] = [];

        for(let j=0;j<ans[i].length;j++)
        {
            arr.push(prev + ans[i][j]);
            prev = ans[i][j];
        }

        let next:number = ans[i].length - 1;
        arr.push(ans[i][next]);
        ans.push(arr);
    }
    return ans;
};

File: 118-pascals-triangle/pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> ans = new ArrayList<>();
        ans.add(new ArrayList<>(Arrays.asList(1)));

        for(int i=0;i<numRows - 1;i++)
        {
            List<Integer> arr = new ArrayList<>();
            int prev = 0;
            int n = ans.get(i).size();

            for(int j=0;j<n;j++)
            {
                arr.add(prev + ans.get(i).get(j));
                prev = ans.get(i).get(j);
            }
            int next = ans.get(i).get(n-1);
            arr.add(next);
            ans.add(arr);
        }

        return ans;
         
    }
}